### ISSUE TITLE
fabric8-auth cannot create spaces on minikube / kubernetes with fabric8

### ISSUE THREAD

**Issue by jstrachan** [fabric8-services/fabric8-auth#91]
so the latest fabric8-auth and fabric8-wit works great on minishift / openshift; but they fail to create spaces on minikube/kubernetes.

The create resource in KeyCloak is falling posting to http://keycloak/auth/realms/fabric8/authz/protection/resource_set with a 403 (or using the public URL too) but failing. e.g.
```json
{
    "POST":"/api/spaces/7a495ffd-223b-4af5-a521-b03eeb555610",
    "action":"Create",
    "ctrl":"SpaceController",
    "from":"172.17.0.20",
    "level":"info",
    "msg":"request started",
    "pkg":"log.LogRequest.func1",
    "req_id":"GYn5i3r7Cq-1852",
    "time":"2017-09-08 14:17:50"
}
{
    "auth_endpoint":"http://keycloak/auth/realms/fabric8/authz/protection/resource_set",
    "file":"/workspace/go/src/github.com/fabric8-services/fabric8-auth/auth/authz.go",
    "func":"github.com/fabric8-services/fabric8-auth/auth.CreateResource",
    "identity_id":"b86be797-4b62-4fe5-8522-df07e2bb28d0",
    "level":"error",
    "line":237,
    "msg":"unable to create a Keycloak resource",
    "pid":1,
    "pkg":"auth",
    "req_headers":
    {
        "Accept-Encoding":
        [
            "gzip"
        ],
        "Authorization":"*****",
        "Content-Length":
        [
            "0"
        ],
        "User-Agent":
        [
            "Go-http-client/1.1"
        ],
        "X-Request-Id":
        [
            "GYn5i3r7Cq-1852"
        ]
    },
    "req_id":"GYn5i3r7Cq-1852",
    "req_params":
    {
        "spaceID":
        [
            "7a495ffd-223b-4af5-a521-b03eeb555610"
        ]
    },
    "resource":
    {
        "name":"7a495ffd-223b-4af5-a521-b03eeb555610",
        "type":"space",
        "scopes":
        [
            "read:space",
            "admin:space"
        ]
    },
    "response_body":"
    {
        \"error\":\"invalid_clientId\",
        \"error_description\":\"Client application 
        [
            fabric8-online-platform
        ]
         is not registered as resource server.\"
    }
    ",
    "response_status":"403 Forbidden",
    "time":"2017-09-08 14:17:50"
}
```

So I guess some work is required to configure `fabric8-online-platform` as a resource server. I've no idea why this fails on minikube but not on minishift though! :)

Anyone got any ideas?

**Comment by alexeykazakov** [comment]
It turned out that AuthZ was disabled for fabric8-platform-online client (despite what is in the config JSON).
@jstrachan can we close this issue now?

**Comment by jstrachan** [comment]
I guess its no longer fabric8-auth issue - but we still need to figure out how to get KeyCloak to not disable those 2 flags on kubernetes - shall we leave this issue here until then?

**Comment by jstrachan** [comment]
Incidentally for anyone who hits this issue, the workaround for now is:
* open the keycloak admin console
* navigate to Clients -> fabric8-online-platform
* enable 'Authorization Enabled' then Save
* click on the new Authorization tab then enable "Remote Resource Management"  then hit Save

**Comment by jstrachan** [comment]
I guess given the oddness we're gonna have to create an init container for keycloak that explicitly tinkers with the REST API to configure things explicitly to avoid using the import JSON mechansim?

**Comment by alexeykazakov** [comment]
Yes, let's keep it open then. We track all authN and authZ issues in this repo even if they are pure KC issues. I added the relevant labels.
This particular issue will not be relevant as soon as we switch to our own authorization implementation (work in progress) instead of using Keycloak for that.
But we still need to solve the problem with proper Keycloak bootstrapping. We have a separate issue for that - https://github.com/fabric8-services/fabric8-auth/issues/92
We could use REST API or Admin CLI to bootstrap KC and stop relaying on JSON config files.

**Comment by jstrachan** [comment]
yeah - we've an issue right now that once someone installs Keycloak we can never again apply any changes to the fabric8 client / KeyCloak as it only imports when there's no resource; so we need a long term replacement for the KeyCloak import mechanism anyway really

**Comment by jstrachan** [comment]
who-hoo! I've figured out a workaround. I've no idea why these flags get removed but when the fabric8-tenant (on kubernetes only) updates the client JSON to add the redirect URI for the new jenkins (so we can use KC to login to the tenant jenkins), for some reason those 2 flags were not present in the JSON; so when we posted the modified JSON back it gets removed. So I've just modified this code to always set those flags when we POST back the JSON to keep then enabled on kubernetes: https://github.com/fabric8-services/fabric8-tenant/pull/325
